src/library/librarycontrol.cpp

```diff
@@ -378,23 +387,38 @@ void LibraryControl::slotMoveFocusForward(double v) {
 
 void LibraryControl::slotMoveFocusBackward(double v) {
     if (v > 0) {
+        refocusLibrary();
         emitKeyEvent(QKeyEvent{QEvent::KeyPress, Qt::Key_Tab, Qt::ShiftModifier});
     }
 }
 
 void LibraryControl::slotMoveFocus(double v) {
     const auto shift = (v < 0) ? Qt::ShiftModifier: Qt::NoModifier;
     const auto times = static_cast<unsigned short>(std::abs(v));
+    refocusLibrary();
     emitKeyEvent(QKeyEvent{QEvent::KeyPress, Qt::Key_Tab, shift, QString(), false, times});
 }
 
 void LibraryControl::emitKeyEvent(QKeyEvent&& event) {
     // Ensure a valid widget has the keyboard focus
     auto focusWidget = QApplication::focusWidget();
+    qDebug() << " ";
+    qDebug() << "   LibraryControl::emitKeyEvent";
+    qDebug() << " 			auto focusWidget = " << focusWidget;
+    qDebug() << " ";
     if (!focusWidget) {
         setLibraryFocus();
         focusWidget = QApplication::focusWidget();
+        qDebug() << " ";
+        qDebug() << " 			if (!focusWidget) {";
+        qDebug() << " 			  setLibraryFocus();";
+        qDebug() << " 			  auto focusWidget = " << focusWidget << "}";
+        qDebug() << " ";
         if (!focusWidget) {
+            qDebug() << " ";
+            qDebug() << " 			  if (!focusWidget) {";
+            qDebug() << " 			    return; }";
+            qDebug() << " ";
```

Comment:
temp debug output. blank lines and indentations so it's easier to spot in the log

Reply:
Qt5 has qTrace() for this purpose. You may also use our simple `kLogger` pattern that adds context to all log lines.

Reply:
@uklotzde I don't find any info on qtrace(), can you point me to some ressources?
Also I don't find a place where `kLogger` usage is explained.